### DISCUSSION TITLE
When phalcon4 ready for PHP8?

### DISCUSSION THREAD

**Discussion by zz** [phalcon/cphalcon#15279] [General]
We need PHP8

**Comment by zz** [comment]
Or phalcon3 for PHP8

**Comment by Jeckerson** [comment]
Hello

You can track progress here:
* https://github.com/phalcon/zephir/issues/2111
* https://github.com/phalcon/cphalcon/issues/15186

List of supported version
| Phalcon version | PHP versions
| --------------------- | --------------------
| 3.4 | 5.6, 7.0, 7.1, 7.2, 7.3
| 4.0, 4.1 | 7.2, 7.3, 7.4
| 4.2 | 7.4, 8.0
| 5.0 | 7.4, 8.0

ETAs
* 4.2 Feb - Mar
* 5.0 - April - May

**Reply by ivastly** [reply]
@Jeckerson thank you for clarification and ETAs.

 I just spent few hours trying to install 4.1 under PHP8, because at [PECL page](https://pecl.php.net/package/phalcon) it clearly says that 4.1 does support PHP8. Which is apprently not true.  

Can we maybe fix the PECL page? Thank you.

**Reply by ivastly** [reply]
![изображение](https://user-images.githubusercontent.com/919655/108275016-f4f8f200-7175-11eb-9c49-a7d96f2b218c.png)

**Reply by Jeckerson** [reply]
@ivastly Hello. It was already fixed, but after we released `4.1`.

https://github.com/phalcon/cphalcon/commit/cb8e962b16d3e8e2f4b58ec796283bbebb4fffc0#diff-37a67ff78eb7260214b323353263b9af40a2fa98719a1e81937fae0159df87a3

Probably, we should release `4.1.1` with that fix...

**Comment by fakharak** [comment]
There is a little confusion. When i say Phalcon or Phalcon 4.2, from both of them i mean CPhalcon for PHP8, not PHP-based Phalcon. 

In some older blog / discussion, i read that Phalcon 5 will be based on PHP (Native) which did not sound good to me as we already have approx dozens of PHP-HMVC/MVC frameworks out there. Why we use CPhalcon (Phalcon based on C) is because it is compiled (compiled HMVC Architecture), and i believe having compiled architecture makes sense even in the presence of JIT of PHP8.

I think most people are looking for CPhalcon (C compiled Phalcon), not PHP-Phalcon (which means not your Phalcon-5). Are we going to get CPhalcon for PHP8 ? I wish we do.

**Reply by michaelbutler** [reply]
Please see this post: https://blog.phalcon.io/post/the-future-of-phalcon due to team members leaving the project due to time constraints, they felt it was not feasible to continue to maintain a C extension for PHP 8.0 and decided to focus their efforts on just being a pure PHP framework in the future (similar to Laravel, Symfony, etc.)

Apparently at the core of the reason is because the Phalcon extension is written in Zephir, which is a PHP-like high level language used to write PHP C extensions more easily; and that project also lacks a team lead now.

That said, yes it looks like Phalcon 4.2 (C extension) is being worked on for PHP 8.0 support.

**Reply by Jeckerson** [reply]
> team member**s** leaving

Only one person left... He was maintaining Zephir part. Now Zephir part is maintained by another people (one of them is me).

Few things left for next cphalcon release:
* Win CI  (https://github.com/phalcon/cphalcon/pull/15329)
* Some minor bugs in Zephir, which are not directly depends on phalcon, but good to fix them now
* Benchmarks part (first of several PRs to check if all good with v4 https://github.com/TechEmpower/FrameworkBenchmarks/pull/6443), after that very curious what benchs will output Phalcon on PHP8
* Docs

So, "Keep calm and wait for PHP8 support".

**Comment by zz** [comment]
Still waiting for Phalcon 4.2.0 release with PHP8 support...
